### PR TITLE
label plots automatically

### LINKED ISSUE #16: Automatically label plots
In response to https://github.com/illinois-ceesd/mirgecom/pull/96#issuecomment-691079491:

It'd be super nice if we could auto-label plot axes with their corresponding expressions.

### PR DESCRIPTION
**Pull request by matthiasdiener**
Fixes #16.

### CODE REVIEW

File: logpyle/runalyzer.py [hunk 1]
```diff
@@ -353,11 +364,15 @@ def execute_magic(self, cmdline):
             from pylab import title
             title(args)
         elif cmd == "plot":
+            import re
+            xlabel, ylabel = re.split(", +|,| +", args)[1:3]
```
Comment (inducer) [inline logpyle/runalyzer.py:368]: This won't work properly half the time. Instead, get the column labels from the DBMS.

Comment (matthiasdiener) [reply logpyle/runalyzer.py:368]: note to self: https://stackoverflow.com/questions/947215/how-to-get-a-list-of-column-names-on-sqlite3-database

Comment (matthiasdiener) [reply logpyle/runalyzer.py]: Done (but required some SQL gymnastics since `mange_sql` destroys the column names).

File: logpyle/runalyzer.py [hunk 2]
```diff
@@ -55,16 +55,19 @@ def get_rank_agg_table(self, qty, rank_aggregator):
...
+
+        plt.xlabel(labels.pop(0) if labels else "")
+        plt.ylabel(labels.pop(0) if labels else "")
 
         if self.interactive:
-            show()
+            plt.show()
 
-    def plot_cursor(self, cursor, *args, **kwargs):
+    def plot_cursor(self, cursor, labels=[], *args, **kwargs):
```
Comment (inducer) [inline logpyle/runalyzer.py:70]:
```suggestion
    def plot_cursor(self, cursor, labels=None, *args, **kwargs):
```
Mutable default arguments are a big no-no.

Comment (inducer) [inline logpyle/runalyzer.py:70]: Should check whether non-`None` labels has correct form and assign X/Y label variables during input normalization.

File: logpyle/runalyzer.py [hunk 3]
```diff
@@ -55,16 +55,19 @@ def get_rank_agg_table(self, qty, rank_aggregator):
         self.rank_agg_tables.add((qty, rank_aggregator))
         return tbl_name
 
-    def scatter_cursor(self, cursor, *args, **kwargs):
-        from matplotlib.pyplot import scatter, show
+    def scatter_cursor(self, cursor, labels=[], *args, **kwargs):
```
Comment (inducer) [inline logpyle/runalyzer.py:58]:
```suggestion
    def scatter_cursor(self, cursor, labels=None, *args, **kwargs):
```
Mutable default arguments are a big no-no.

File: logpyle/runalyzer.py [hunk 4]
```diff
@@ -353,11 +359,15 @@ def execute_magic(self, cmdline):
...
+            columnnames = [column[0] for column in cursor.description]
             self.db.plot_cursor(self.db.db.execute(
-                self.db.mangle_sql(args)))
+                self.db.mangle_sql(args)), columnnames)
         elif cmd == "scatter":
+            cursor = self.db.db.execute(self.db.mangle_sql(args))
+            columnnames = [column[0] for column in cursor.description]
             self.db.scatter_cursor(self.db.db.execute(
-                self.db.mangle_sql(args)))
+                self.db.mangle_sql(args)), columnnames)
```
Comment (inducer) [inline logpyle/runalyzer.py:370]: These run the query twice, which seems wasteful. Why?

Comment (matthiasdiener) [reply logpyle/runalyzer.py]: Fixed.

File: logpyle/runalyzer.py (hunk 3 above)
Comment (matthiasdiener) [reply logpyle/runalyzer.py]: Fixed.

File: logpyle/runalyzer.py (hunk 2 above)
Comment (matthiasdiener) [reply logpyle/runalyzer.py]: Fixed.

Comment (matthiasdiener) [reply logpyle/runalyzer.py]: Not 100% what you had in mind here but I implemented something in this direction.

**Comment by matthiasdiener** [comment]
Needs a review

**Review by inducer — COMMENTED** [review]
Thanks. A few more, good to go after that.

File: logpyle/runalyzer.py [hunk 5]
```diff
@@ -166,10 +181,10 @@ def replace_magic_column(match):
             if rank_aggregator is not None:
                 rank_aggregator = rank_aggregator[1:]
                 magic_columns.add((qty_name, rank_aggregator))
-                return f"{rank_aggregator}_{qty_name}.value"
+                return f"{rank_aggregator}_{qty_name}.value AS {qty_name}"
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply logpyle/runalyzer.py:184]: Added a simple test.

**Review by inducer — APPROVED** [review]
Thx!

**Pull request merged**